Some unit tests crash on Python 2.7 using pytest 4.2.1

##### SUMMARY

Some unit tests crash on Python 2.7 using pytest 4.2.1.

##### ISSUE TYPE

Bug Report

##### COMPONENT NAME

unit tests

##### ANSIBLE VERSION

```
ansible 2.8.0.dev0
  config file = None
  configured module search path = [u'/Users/mclay/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/mclay/code/mattclay/ansible/lib/ansible
  executable location = /Users/mclay/code/mattclay/ansible/bin/ansible
  python version = 2.7.14 (default, Mar 22 2018, 11:39:16) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

N/A

##### STEPS TO REPRODUCE

Update the version constraint for pytest to: `pytest == 4.2.1 ; python_version >= '2.7'`

https://github.com/ansible/ansible/blob/02c9f708981462302245d7ae760499b90ec49b14/test/runner/requirements/constraints.txt#L13

Run the unit tests that crash:

`ansible-test units test/units/module_utils/basic/test_atomic_move.py test/units/module_utils/basic/test_set_mode_if_different.py --docker -v --python 2.7`

##### EXPECTED RESULTS

Tests pass.

##### ACTUAL RESULTS

```
=================================== FAILURES ===================================
______________ test_rename_perms_fail_temp_succeeds[stdin1-False] ______________
[gw3] linux2 -- Python 2.7.12 /usr/bin/python2.7
:-1: running the test CRASHED with signal 0
------------------------------- captured stderr --------------------------------
/usr/lib/python2.7/genericpath.py:40: AttributeError: 'list' object has no attribute 'st_mode'
______________ test_rename_perms_fail_temp_succeeds[stdin0-True] _______________
[gw2] linux2 -- Python 2.7.12 /usr/bin/python2.7
:-1: running the test CRASHED with signal 0
------------------------------- captured stderr --------------------------------
/usr/lib/python2.7/genericpath.py:40: AttributeError: 'list' object has no attribute 'st_mode'
___________________ test_missing_lchmod_is_link[True-stdin0] ___________________
[gw0] linux2 -- Python 2.7.12 /usr/bin/python2.7
:-1: running the test CRASHED with signal 0
------------------------------- captured stderr --------------------------------
/usr/lib/python2.7/posixpath.py:414: OSError: [Errno 22] Invalid argument: '/root'
__________________ test_missing_lchmod_is_link[False-stdin1] ___________________
[gw3] linux2 -- Python 2.7.12 /usr/bin/python2.7
:-1: running the test CRASHED with signal 0
------------------------------- captured stderr --------------------------------
/usr/lib/python2.7/posixpath.py:414: OSError: [Errno 22] Invalid argument: '/root'
--- generated xml file: /root/ansible/test/results/junit/python2.7-units.xml ---
=========================== short test summary info ============================
FAILED test/units/module_utils/basic/test_atomic_move.py::test_rename_perms_fail_temp_succeeds[stdin1-False]
FAILED test/units/module_utils/basic/test_atomic_move.py::test_rename_perms_fail_temp_succeeds[stdin0-True]
FAILED test/units/module_utils/basic/test_set_mode_if_different.py::test_missing_lchmod_is_link[True-stdin0]
FAILED test/units/module_utils/basic/test_set_mode_if_different.py::test_missing_lchmod_is_link[False-stdin1]
```

--- a/test/units/module_utils/basic/test_atomic_move.py
+++ b/test/units/module_utils/basic/test_atomic_move.py
@@ -197,7 +197,7 @@ def test_rename_perms_fail_temp_succeeds(atomic_am, atomic_mocks, fake_stat, moc
     mock_context = atomic_am.selinux_default_context.return_value
     atomic_mocks['path_exists'].return_value = False
     atomic_mocks['rename'].side_effect = [OSError(errno.EPERM, 'failing with EPERM'), None]
-    atomic_mocks['stat'].return_value = [fake_stat, fake_stat, fake_stat]
+    atomic_mocks['stat'].return_value = fake_stat
     atomic_mocks['stat'].side_effect = None
     atomic_mocks['mkstemp'].return_value = (None, '/path/to/tempfile')
     atomic_mocks['mkstemp'].side_effect = None

--- a/test/units/module_utils/basic/test_set_mode_if_different.py
+++ b/test/units/module_utils/basic/test_set_mode_if_different.py
@@ -151,3 +151,6 @@ def test_missing_lchmod_is_link(am, mock_stats, mocker, check_mode):
         assert not m_chmod.called
     else:
         m_chmod.assert_called_with(b'/path/to/file/no_lchmod', 0o660)
+
+    mocker.resetall()
+    mocker.stopall()

